docs/configuration/initialization.md

```diff
@@ -0,0 +1,129 @@
+---
+id: initialization
+title: Initialization
+---
+
+In Next.js, you can define an API route that will catch all requests that begin with a certain path. Conveniently, this is called [Catch all API routes](https://nextjs.org/docs/api-routes/dynamic-api-routes#catch-all-api-routes).
+
+When you define a `/pages/api/[...nextauth]` JS/TS file, you instruct NextAuth.js that every API request beginning with `/api/auth/*` should be handled by the code written in the `[...nextauth]` file.
+
+Depending on your use-case, you can initialize NextAuth.js in two different ways:
+
+## Simple initialization
+
+In most cases, you won't need to worry about what `NextAuth.js` does, and you will get by just fine with the following initialization:
+
+```ts title="/pages/api/auth/[...nextauth].js"
+import NextAuth from "next-auth"
+
+export default NextAuth({
+  ...
+})
+```
+
+Here, you only need to pass your [options](/configuration/options) to `NextAuth`, and `NextAuth` does the rest.
+
+This is the preferred initialization in tutorials/other parts of the documentation, as it somewhat simplifies the code and there is smaller chances for errors in the authentication flow.
+
+## Advanced initialization
+
+In rare cases, there is just **some extra functionality** that you wish NextAuth.js had, or you need to _preprocess_ the request, but you understand that it is way too specific for your use-case to be introduced as part of the library.
```

Comment:
I would tighten up this intro part a bit, maybe something along the lines of - "If you have a specific use case and need to make NextAuth.js do something slightly different than what its designed for, keep in mind, the `[...nextauth].js` config file is still just **a regular [API Route]()** at the end of the day.."